Mindbox.Analyzers.Tests.Framework/ClassMethodCallProhibitedRuleTests.cs

```diff
@@ -0,0 +1,497 @@
+                        .Location), // Load System.Private.CoreLib into test compilation
+                MetadataReference.CreateFromFile(
+                    typeof(Console).Assembly.Location), // Load System.Console into test compilation
+                MetadataReference.CreateFromFile(
+                    Assembly.Load("System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a")
+                        .Location)) // Load System.Runtime into test compilation
+            .AddSyntaxTrees(syntaxTree);
+
+        var diagnostics = compilation.GetDiagnostics().Where(d => d.Severity == DiagnosticSeverity.Error).ToArray();
+        if (diagnostics.Length > 0)
+        {
+            throw new Exception(
+                $"There were {diagnostics.Length} compile errors:\n{string.Join("\n", diagnostics.Select(d => d.ToString()))}");
+        }
+
+        return compilation.GetSemanticModel(syntaxTree);
+    }
+
+    #region MAHC Single file tests
+
+    [TestMethod]
+    public void MAHC_NoCalls_DoesntChangeSourceCode()
+    {
+        const string givenSource = @"using System;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        Console.WriteLine(""No MAHC calls!"");
+    }
+}";
+        var semanticModel = GetSemanticModelFromSourceCode("Program.cs", givenSource);
+        _mahcRule.AnalyzeModel(semanticModel, out var problems);
+
+        Assert.AreEqual(0, problems.Count);
+    }
+
+    [TestMethod]
+    public void MAHC_OneSingleLineCall_WrapsInPragma()
+    {
+        const string givenSource = @"using System;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var service = Itc.Commons.Model.ModelApplicationHostController.Instance.Get<Itc.Commons.Model.ITenantValidator>();
+    }
+}";
+
+        const string filename = "Program.cs";
+
+        var semanticModel = GetSemanticModelFromSourceCode(filename, givenSource);
+        _mahcRule.AnalyzeModel(semanticModel, out var problems);
+
+        Assert.AreEqual(1, problems.Count);
+        var problem = problems.Single();
+        Assert.AreEqual(_mahcRule.DiagnosticDescriptor.Id, problem.Id);
+        Assert.AreEqual(6, problem.Location.GetLineSpan().StartLinePosition.Line); // LinePosition.Line is 0-based
+        Assert.AreEqual(6, problem.Location.GetLineSpan().EndLinePosition.Line);
+    }
+
+    [TestMethod]
+    public void MAHC_TwoSeparatedSingleLineCalls_WrapsInTwoPragmas()
+    {
+        const string givenSource = @"using System;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var service1 = Itc.Commons.Model.ModelApplicationHostController.Instance.Get<Itc.Commons.Model.ITenantValidator>();
+        var separator = 0;
+        var service2 = Itc.Commons.Model.ModelApplicationHostController.Instance.Get<Itc.Commons.Model.ITenantValidator>();
+    }
+}";
+
```

Comment:
Тесты еще осталось подправить, теперь непонятно, зачем тестировать несколько файлов или несколько строк. Я бы потестировал без MAHC, с и MAHC в отдельной переменной

Reply:
в правилах действительно можно избавиться от тестов нескольких файлов, а в консольке я так баг отловил :)

насчёт тестов с запретными классами в переменной хорошая идея, странно что я её не имплементировал, я же тестил так, когда проверял, что работает.

сто́ит ли ещё проверять, когда пассим манс как параметр метода?

Reply:
давай проверим, хорошая идея

Reply:
Думаю, мажорная в данном кейсе норм.

1. Тебе не нужен суфкис дев. Он добавится автоматически, когда смержишь бранч в ветку publish-dev (https://github.com/mindbox-moscow/Mindbox.Analyzers/blob/main/.github/workflows/publish-dev.yml). А после мержа в мастер уже обновится основная версия пакета.
1. У тебя все еще остались лишние тесты, например какой смысл в MAHC_TwoSeparatedSingleLineCalls_ProducesTwoProblems и в MAHC_TwoConsecutiveSingleLineCalls_ProducesTwoProblems? Какая разница SingleLine или MultiLine? Все еще торчит прагма NOMC_OneSingleLineCall_WrapsInPragma.
1. Сори, только заметил, непонятно, зачем в одном классе тестировать 2 правила сразу давай разделим тесты? Ну и приватный метод должен быть внизу по всем понятиям кодстайла)

Reply:
1. [Remove the dev suffix because it will be added automatically](https://github.com/mindbox-moscow/Mindbox.Analyzers/pull/6/commits/7caa296c3fcf032ebb155b122104408e9323dfcb)
2. [Rename remaining tests](https://github.com/mindbox-moscow/Mindbox.Analyzers/pull/6/commits/a98333e85ad60541820ef0bbb06b58bc9b236688), [Remove redundant tests](https://github.com/mindbox-moscow/Mindbox.Analyzers/pull/6/commits/cf49ab9647b5ae4c229f13cd95e72e2383c6526f)
3. [Split up tests for rules](https://github.com/mindbox-moscow/Mindbox.Analyzers/pull/6/commits/41fa4e46df0ee4a6c10ec18275fc83bf92c0085c)

исправил)

Reply:
@itsyrulnikov тегну тут на всякий)

Reply:
Все еще не понимаю, зачем нужны тесты на MultiLine, но черт с ним, давай уже мержить в publish-dev и тестировать анализатор, а потом консольку накатывать.

Reply:
@GureevLeonid апрувни пж, не могу вмерджить

Reply:
@itsyrulnikov а я в main ПР создал, мне пересоздать ПР в ветку publish-dev?

Reply:
не через PR, просто смержи гитом локально, вряд ли там PR обязателен